### DISCUSSION TITLE
How exactly does NuxtPage work?

### DISCUSSION THREAD

**Discussion by Thomasparsley** [nuxt/nuxt#25322] [Questions]
Hello,

I want to replicate the functionality of the vue-router. The functionality is that I want to have a default and a toolbar in components for one path.
Default is the page component itself and toolbar is the toolbar component.

But nuxt assigns the page path to the router as file.

My question is, can such logic be replicated using NuxtPage? So how exactly does NuxtPage work when it wraps a router-view inside itself?

I have tried using name with NuxtPage, but have not been able to achieve the desired logic.

**Comment by xjccc** [comment]
i think you should use `layout`?
use layout name to change?
```
// app.vue
 <NuxtLayout>
    <NuxtPage />
  </NuxtLayout>
```

```
// layout - default
<template>
<slot />
<toolbar />
</template>

```

**Reply by Thomasparsley** [reply]
is it possible to have layout in layout?

**Reply by xjccc** [reply]
you can change layout in your page
```
definePageMeta({
  layout: 'custom'
})
```
